tests/unit/snap/test_build.py

```diff
@@ -0,0 +1,277 @@
+# This file is part of craft-platforms.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for snap-specific platforms functions."""
+
+import craft_platforms
+import pytest
+import pytest_check
+from craft_platforms.snap import _build
+from hypothesis import given, strategies
+
+
+@given(strategies.integers(min_value=16, max_value=98).filter(lambda x: x % 2 == 0))
+def test_core_base_regex_match(version):
+    assert _build.CORE_BASE_REGEX.match(f"core{version}")
+
+
+@given(
+    strategies.one_of(
+        strategies.integers(min_value=99),
+        strategies.integers(min_value=16, max_value=98).filter(lambda x: x % 2 != 0),
+        strategies.integers(max_value=15),
+    ),
+)
+def test_core_base_regex_version_non_match(version):
+    assert not _build.CORE_BASE_REGEX.match(f"core{version}")
+
+
+@given(strategies.text().filter(lambda s: not s.startswith("core")))
+def test_core_base_regex_non_match(string):
+    assert not _build.CORE_BASE_REGEX.match(string)
+
+
+@pytest.mark.parametrize(
+    ("string", "matches"),
+    [
+        ("core", True),
+        ("core14", False),
+    ],
+)
+def test_core_base_regex_match_specials(string, matches):
+    assert bool(_build.CORE_BASE_REGEX.match(string)) == matches
+
+
+@pytest.mark.parametrize(
+    ("base", "expected"),
+    [
+        ("core", craft_platforms.DistroBase("ubuntu", "16.04")),
+        ("core16", craft_platforms.DistroBase("ubuntu", "16.04")),
+        ("core18", craft_platforms.DistroBase("ubuntu", "18.04")),
+        ("core20", craft_platforms.DistroBase("ubuntu", "20.04")),
+        ("core22", craft_platforms.DistroBase("ubuntu", "22.04")),
+        ("core24", craft_platforms.DistroBase("ubuntu", "24.04")),
+        ("core26", craft_platforms.DistroBase("ubuntu", "26.04")),
+        ("devel", craft_platforms.DistroBase("ubuntu", "devel")),
+    ],
+)
+def test_get_distro_base_from_core_base_success(base, expected):
```

Comment:
Can you add a scenario for `base: core22-desktop`?  I think your implementation will support it.

Reply:
Added. I did have to change the implementation slightly, but it's only an improvement.

Reply:
Using `base: core22-desktop` requires `build-base: core22`, right?

Reply:
As of now, yes.  `-desktop` is missing from the ST119 spec, so I'm not sure if that's intentional or not.  What do you think, @cmatsuoka?

Reply:
The `-desktop` variants should make sense only for runtime bases, the build bases will be the same AFAIK.